t/t0017-security.t

```diff
@@ -282,6 +285,27 @@ test_expect_success 'dispatcher suppresses guest event to same guest connection'
 	! grep -q test.a ev8.out
 '
 
+test_expect_success HAVE_JQ 'guests may add userid-prefixed services' '
+	USERID=$(id -u) &&
+	${jq} -n "{service: \"${USERID}-sectest\"}" >user_service.json &&
+	FLUX_HANDLE_ROLEMASK=0x2 ${RPC} service.add <user_service.json
+'
+
+test_expect_success HAVE_JQ 'guests may not add other-userid-prefixed services' '
+	USERID=$(($(id -u)+1)) &&
+	${jq} -n "{service: \"${USERID}-sectest\"}" >user2_service.json &&
+	! FLUX_HANDLE_ROLEMASK=0x2 ${RPC} service.add <user2_service.json \
```

Comment:
One thing here I just noticed. Use of `!` in sharness scripts is discouraged, since the test could "pass" even if the command segfaults. However, to get this to work and pass in an environment variable, you'd have to do something like:
```sh
        ( export FLUX_HANDLE_ROLEMASK=0x2 &&
          test_expect_code 1 \
            ${RPC} service.add <user2_service.json \
              2>uservice_add.err
        ) &&
```

Which I can understand is annoying. So maybe just for future reference...

Reply:
Ah I just made a mental note of that pattern in your flux-security tests.  Thanks!  Will fix.